lightning-invoice/src/payment.rs

```diff
@@ -466,16 +603,46 @@ where
 		}.map_err(|e| PaymentError::Sending(e))
 	}
 
+	// Takes in a path to have its information stored in `payment_cache`. This is done for paths
+	// that are pending retry.
+	fn process_path_inflight_htlcs(&self, payment_hash: PaymentHash, hops: Vec<RouteHop>) {
+		let payment_info_route_hops: Vec<PaymentInfoRouteHop> = hops.iter().map(|h| PaymentInfoRouteHop { pubkey: h.pubkey, short_channel_id: h.short_channel_id, fee_msat: h.fee_msat }).collect();
+
+		self.payment_cache.lock().unwrap().entry(payment_hash)
+			.or_insert(PaymentInfo {
+				attempts: PaymentAttempts {
+					count: 1,
+					first_attempted_at: T::now()
+				},
+				paths: vec![]
+			})
+			.paths.push(payment_info_route_hops);
```

Comment:
Why not use `PaymentInfo`? Won't the `count` field be inaccurate when this is called with `Ok` results? I think we want it to be zero in all cases actually and let `retry_payment` increment the count.

Reply:
ah, yes, that makes sense.